### ISSUE TITLE
Support SOVERSION-ed libraries

### ISSUE THREAD

**Issue by oleksandr-pavlyk** [scikit-build/scikit-build#625]
On Linux, for projects that set `VERSION`/`SOVERSION` properties on the dynamic library they install CMake generates symbolic links symbolic links for least specified version to the most specified one:

```
(work) [16:53:47 vmbox dpctl]$ ls -l _skbuild/linux-x86_64-3.9/cmake-install/dpctl/libDP*
_skbuild/linux-x86_64-3.9/cmake-install/dpctl/libDPCTLSyclInterface.so -> libDPCTLSyclInterface.so.0
_skbuild/linux-x86_64-3.9/cmake-install/dpctl/libDPCTLSyclInterface.so.0 -> libDPCTLSyclInterface.so.0.12
_skbuild/linux-x86_64-3.9/cmake-install/dpctl/libDPCTLSyclInterface.so.0.12
```

scikit-build copies these symbolic links as files, because it uses `shutils.copyfile` with `follow_symlinks=True` default (see [setuptools_wrap.py#L855-L856](https://github.com/scikit-build/scikit-build/blob/master/skbuild/setuptools_wrap.py#L855-L856)). Each file copied into Python layout is a regular file, and this interferes with proper working of the library.

I have fixed this issue locally, by modifying these 2 lines to 

```python
    copyfile(src_file, dest_file, follow_symlinks=False)
    copymode(src_file, dest_file, follow_symlinks=False)
```

**Comment by henryiii** [comment]
Would this interfere with producing wheels, though? Wheels do not allow symlinks, since they are internally zip files, and Python (still) has a standard library bug that keeps it from supporting symlinks in zips.

**Comment by oleksandr-pavlyk** [comment]
@henryiii I trust you are aware of https://github.com/cournape/zipfile2 package, which adds metadata to zip files to support read/write of symbolic links.

**Comment by henryiii** [comment]
No, given there are only three stars on it, I'm not sure it's common knowledge. And I'm pretty sure pip, pipx, wheel, poetry, pdm, etc. all use the standard lib to work with wheels, not zipfile2, so it would be of no help loading the wheels anywhere.

**Comment by henryiii** [comment]
scikit-build (classic) after #1185 is now a thin wrapper around scikit-build-core's setuptools plugin. Closing all open issues; if this still applies, please reopen or open an issue on scikit-build-core. Thanks!
